src/main/java/org/tugraz/sysds/runtime/data/DenseBlockFP32.java

```diff
@@ -149,7 +149,7 @@ public DenseBlock set(int r, int c, double v) {
 	
 	@Override
 	public DenseBlock set(DenseBlock db) {
-		System.arraycopy(db.valuesAt(0), 0, _data, 0, _rlen*_odims[0]);
+		System.arraycopy(DataConverter.toFloat(db.valuesAt(0)), 0, _data, 0, _rlen*_odims[0]);
```

Comment:
Again maybe in a different PR (perf), I'd like to avoid unnecessary array copies (toFloat()) just to use an array copy after that. Probably the best would be to have a stream/loop to set all elements by using get double and a special case for common input/output arrays.

Reply:
I added the Todos for performance, I guess this resolves this for now. Same for the DenseBlockBool